File: damon-example/src/main/java/com/zyhang/damon/example/MainActivity.java
package com.zyhang.damon.example;

import android.os.Bundle;
import android.util.Log;

import com.zyhang.damon.RequiresPresenter;
import com.zyhang.damon.rxjava.support.MvpAppCompatActivityRx;

@RequiresPresenter(MainPresenter.class)
public class MainActivity extends MvpAppCompatActivityRx<MainPresenter> implements MainView {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    public void log(String tips) {
        Log.i("main", tips);
    }
}

File: damon-example/src/main/kotlin/com/zyhang/damon/example/MainActivity.kt
package com.zyhang.damon.example

import android.content.Intent
import android.os.Bundle
import android.view.View
import com.zyhang.damon.rxjava.kotlin.EmptyPresenterRxMvpAppCompatActivity

/**
 * Created by zyhang on 2018/8/10.10:30
 */

@Suppress("UNUSED_PARAMETER")
class MainActivity : EmptyPresenterRxMvpAppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


    }

    fun single(view: View) {
        startActivity(Intent(this, SinglePresenterActivity::class.java))
    }

    fun multi(view: View) {
        startActivity(Intent(this, MultiPresenterActivity::class.java))
    }
}